src/packages/galaxy/galaxy.py

```diff
@@ -67,12 +67,13 @@ def __init__(self, url: str = None):
             A ``StarSystem`` object representing the found system, or ``None`` if none was found.
         """
 
-        data = await self._call("search", {"name": name, "limit": 1})
-        if 'data' in data and data['data'] and data['data'][0]['name'].casefold() == name.casefold():
+        data = await self._call("api/systems", {"filter[name:ilike]": name, "sort": "name", "limit": 1})
```

Comment:
Is the intention to use a different function for !search and other wilder-matching commands? Using the JSONAPI endpoint limits you to the single search strategy (ILIKE, without wildcards), while the /mecha endpoint is designed for mecha's !search functionality, providing similarity based searches.

Reply:
The intention here is to fetch a system by name, case insensitively. Any system given to this method is expected to at least have the correct spelling, the call to the API is simply used to verify the correct capitalization. We have a separate `search_systems_by_name` method that uses `/mecha` for its results.